chore(doc): add troubleshooting doc for virtual disk blockdevices (#30)
openebs/upgrade: docs/virtual-disk-troubleshoot.md

  ```sh
  $ kubectl edit spc cstor-pool
  apiVersion: openebs.io/v1alpha1
  kind: StoragePoolClaim
  metadata:
    annotations:
      reconcile.openebs.io/disable: "true"
      cas.openebs.io/config: |
        - name: PoolResourceRequests
          value: |-
              memory: 2Gi
        - name: PoolResourceLimits
          value: |-
              memory: 4Gi
    creationTimestamp: "2020-08-03T10:45:22Z"
    finalizers:
    - storagepoolclaim.openebs.io/finalizer
  .......
  .......
  ```
  After adding the above annotation SPC will generate the following events on it. Which means changes made to SPC will not be reconciled until annotation is removed. Verify events using following command
  ```sh
   $ k describe spc cstor-pool 
   Events:
    Type     Reason  Age               From            Message
    ----     ------  ----              ----            -------
    Warning  Update  7s (x2 over 13s)  spc-controller  reconcile is disabled via "reconcile.openebs.io/disable" annotation
  
  ```

> - Update the CSP with correct blockdevice details. Replace the old blockdevices with new blockdevices.

Reviewer: ```suggestion
- Update the CSP with correct blockdevice details. Replace the old blockdevices with new blockdevices and dev links also needs to be updated.
```

        name: blockdevice-99596b3a4eb2396b8e45b334daeefccc
  ```

- Update the blockdevice entry on SPC. This step is required to make it intact with CSP since we manually updated the blockdevice name in CSP.
  ```sh
  $ kubectl edit spc cstor-pool
  apiVersion: openebs.io/v1alpha1
  kind: StoragePoolClaim
  metadata:
    annotations:
      reconcile.openebs.io/disable: "true"
    creationTimestamp: "2020-08-03T10:45:22Z"
    finalizers:
    - storagepoolclaim.openebs.io/finalizer
    .....
    .....
  spec:
    blockDevices:
      blockDeviceList:
      - blockdevice-8e782e6b47ed896a325870fc436e65f9
      - blockdevice-99596b3a4eb2396b8e45b334daeefccc
  ```
  After editing the SPC we are making sure that CSP and SPC are pointing to the same blockdevices and cStor pool is also running on top of mentioned devices. Verify whether it saved successfully or not using following command
  ```sh
  $ kubectl get spc cstor-pool -o yaml | grep blockdevice-
      - blockdevice-8e782e6b47ed896a325870fc436e65f9
      - blockdevice-99596b3a4eb2396b8e45b334daeefccc
  ```

- Enable the reconciliation on SPC. This step is required to enable back the reconciliation on SPC which was stopped by

> adding `reconcile.openebs.io/disable annotation` on SPC. Enabling can be achieved by removing annotation on SPC using `kubectl edit` command. Then remove the annotation `reconcile.openebs.io/disable` which was added in above steps. This will help to reconcile further changes in SPC if made. After removing the annotation there shouldn’t be any more events generated for SPC saying disabled reconciliation.

Reviewer: After modifying will it be good to rerun the script before upgrading/migrating?

Author: IMO it is not required as after upgrade this situation will not occur.

# Troubleshooting Virtual Disk Blockdevices

When using OpenEBS with virtual disks for blockdevices, there can be scenarios where the disk is detached and reattached to the node. This may result in the creation of new blockdevices but the underlying disk remains the same. This issue is fixed in OpenEBS 2.0.0 onwards. Before migrating SPC to CSPC or upgrading OpenEBS from 1.12.0 or below to 2.0.0 or above follow the steps below.

## Steps to update CStor pool with correct blockdevices

> - Run the diagnostic [script](verifybds.sh) that will identify if any blockdevice is renamed after reattachment. The script takes `<openebs-namespace>` as the only input.

Reviewer 2: ```suggestion
- Run the diagnostic [script](verify_cstor_bds.sh) that will identify if any blockdevice is renamed after reattachment. The script takes `<openebs-namespace>` as the only input.
```

  ```sh

>   $ ./verifybds.sh openebs

Reviewer 2: ```suggestion
  $ ./verify_cstor_bds.sh openebs
```

  ```sh
  $ kubectl -n openebs get bd blockdevice-8e782e6b47ed896a325870fc436e65f9 --show-labels 
  NAME                                           NODENAME                                       SIZE          CLAIMSTATE   STATUS   AGE   LABELS
  blockdevice-8e782e6b47ed896a325870fc436e65f9   ip-192-168-20-104.us-east-2.compute.internal   10737418240   Claimed      Active   64m   kubernetes.io/hostname=ip-192-168-20-104,ndm.io/blockdevice-type=blockdevice,ndm.io/managed=true
  ```
  This will help us identify the hostname for the blockdevice from the `kubernetes.io/hostname` label. Fill in the below template and apply it.
  ```yaml
  apiVersion: openebs.io/v1alpha1
  kind: BlockDeviceClaim
  metadata:
    finalizers:
    - storagepoolclaim.openebs.io/finalizer
    labels:
      openebs.io/storage-pool-claim: <spc-name>
    name: bdc-<bd-name>
    namespace: <openebs-namespace>
    ownerReferences:
    - apiVersion: openebs.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: StoragePoolClaim
      name: <spc-name>
      uid: <spc-uuid>
  spec:
    blockDeviceName: <bd-name>
    blockDeviceNodeAttributes:
      hostName: <host-name>

>     hostName: <host-name>

Reviewer 2: ```suggestion
```

  ```
  Check the BDC status using the command 
  ```sh
  $ kubectl -n <openebs-namespace> get bdc <bdc-name>
  ```
  If the BD is not claimed automatically then check the BD for the `internal.openebs.io/uuid-scheme` annotation. If it is set to `legacy` then
  - add the label `openebs.io/block-device-tag: <spc-name>` and remove the annotation `internal.openebs.io/uuid-scheme`.
  - add the selector below to the BDC created using above yaml.
    ```yaml
    selector:
      matchLabels:

>         openebs.io/block-device-tag: <spc-name>

Reviewer 2: :+1:
